### DISCUSSION TITLE
How to select another input stream on the go?

### DISCUSSION THREAD

**Discussion by vorosj** [pschatzmann/arduino-audio-tools#1609] [Q&A] [answered]
For example I have several possible input streams (adc, generator, Bluetooth a2dp), and a single i2s output.
How can I select the input on the go? I do not need mixing, and don't want to load the uC with the unused streams.

**Comment by pschatzmann** [comment, marked as answer]
You can dynamically reconfigure the copier with the help of the begin method.

**Reply by vorosj** [reply]
Thank you.
For example I start a Bluetooth and a generator input stream, and an i2s output stream. 
If I first configure the copier as a2dp->i2s, then reconfigure copier to generator->i2s, the a2dp task is still receiving and decoding, or it is just sitting in the memory (not using notable computing time)?

**Reply by pschatzmann** [reply]
Yes it does: I would suggest to call the avrc commands pause() and play() (or maybe set_connected(false) and set_connected(true) ?)

**Comment by vorosj** [comment]
Just a story of a finished project:
I built a hearing aid device for our 101 years old grandmother. :-) It was difficult for her to understand conversation using her normal hearind aid device. But she understand very well the TV using a big headset. 
So I built a Bluetooth wireless mic using your great library for a big JBL headset. If we keep the mic close, the sound is clear, without the disturbing noise which she often complained about using the normal hearing aid . 
She really likes the new one...
